### PR TITLE
@alloy - Truncate artwork title with ellipsis 

### LINKED ISSUE #160: Tag artwork brick title should truncate so they don't wrap multiple lines
screen shot 2017-05-16 at 11 05 36 am

### PR DESCRIPTION
**Pull request by broskoski**
Closes #160 

Pro tip: `min-width: 0` is the magic property on a flex container to have `text-overflow` work properly....the more you know.

### CODE REVIEW

File: src/components/artwork/details.tsx [hunk 1]
```diff
@@ -25,8 +25,14 @@ export class ArtworkDetails extends React.Component<DetailsProps, null> {
...
   titleLine() {
+    const titleStyle = {
+      display: "block",
+      textOverflow: "ellipsis",
+      overflow: "hidden",
+      whiteSpace: "nowrap",
+    }
     return (
-      <TextLink href={this.props.artwork.href}>
+      <TextLink href={this.props.artwork.href} style={titleStyle}>
```
Comment (alloy) [inline src/components/artwork/details.tsx]: Why is this an inline style vs a styled component?

File: src/components/text_link.tsx [hunk 2]
```diff
@@ -6,7 +6,8 @@ import "../assets/fonts"
 export interface LinkProps extends React.Props<TextLink>, React.HTMLAttributes<TextLink> {
   href?: string
   underline?: boolean
-  color?: string
+  color?: string,
+  style?: any,
```
Comment (alloy) [inline src/components/text_link.tsx]: Regardless of the question why this uses inline styling, I’m surprised you’d need to specify this property and it should probably be of the [CSSProperties](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/299cbfa81a20929b0362815730239f9dcc992449/types/react/index.d.ts#L648) type.

File: src/components/artwork/details.tsx (hunk 1 above)
Comment (broskoski) [inline src/components/artwork/details.tsx]: Good call, willfix

**Comment by alloy** [comment]
Gotta 📸

**Comment by broskoski** [comment]
screenshot 2017-05-16 18 34 00

**Comment by alloy** [comment]
Perfect composition 👍

**Pull request merged**